nizkctf/cli/news.py

```diff
@@ -0,0 +1,93 @@
+# -*- encoding: utf-8 -*-
+
+from __future__ import unicode_literals, division, print_function,\
+                       absolute_import
+import os
+import sys
+import time
+import json
+import pysodium
+from base64 import b64decode
+from ..news import News
+from ..team import my_team
+from ..text import width
+
+from .teamsecrets import TeamSecrets
+from ..team import Team
+
+def pprint(news, team_only):
+    '''
+    Pretty print news in terminal.
+
+    Args:
+        news (list): List of news.
+        team_only (bool): whether to display only team directed news or not.
+    '''
+
+    if len(news) == 0:
+        print('No news yet.')
+        return
+
+    team = my_team()
+
+    # FIXME improve filtering
+    team_news = [n for n in news if n.get('to') == team['name']]
+    if team_only:
+        news = team_news
+    else:
+        news = [n for n in news if 'to' not in n] + team_news
+    
+    def decrypt_msg(msg):
+        team_pk, team_sk = team['crypt_pk'], TeamSecrets['crypt_sk']
+        
+        # TODO Check if that's the better approach
+        # reference: https://download.libsodium.org/doc/public-key_cryptography/sealed_boxes.html
+        return pysodium.crypto_box_seal_open(msg, team_pk, team_sk)
+
+    def decode_news(news):
+        # News messages are always in base 64
+        news['msg'] = b64decode(news['msg'])
+        if 'to' in news:
+            # Message was sent to a team, so we need to decrypt it
+            news['msg'] = decrypt_msg(news['msg'])
+
+            try:
+                # TODO this fails in case the message was not encoded before
+                #      being encrypted. Change accordingly to the news add behaviour
+                news['msg'] = b64decode(news['msg'])
+            except:
+                pass
+
+        news['msg'] = news['msg'].decode('utf-8')
```

Comment:
this should be needed only for encrypted messages (inside the "if" above)

for decrypted messages, it would probably fail in Python3 with
AttributeError: 'str' object has no attribute 'decode'